src/averageAges.js

```diff
@@ -55,7 +65,27 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(person =>
+    people.some(mom => (person.name === mom.mother && person.sex === 'f')));
+  const children = people.filter(person =>
+    mothers.some(mom => (person.mother === mom.name)));
+  const onlySon = children.filter(son => son.sex === 'm');
+
+  function getAllAge(humans) {
+    return humans.map(child =>
+      child.born - mothers.find(mom => child.mother === mom.name).born);
+  };
+
+  function averageAge(human) {
+    return human.reduce((sum, age) => sum + age, 0)
+      / human.length;
+  };
+
+  const allAgeChildren = getAllAge(children);
+  const allAgeOnlySon = getAllAge(onlySon);
+
+  return onlyWithSon ? averageAge(allAgeOnlySon)
```

Comment:
Replace the ternary operator with a single call to the `average` function and pass the appropriate array as an argument. Example: `return average(onlyWithSon ? allAgeOnlySon : allAgeChildren);`